packages/eas-cli/src/project/ios/bundleIdentifier.ts

```diff
@@ -28,22 +28,38 @@ export async function ensureBundleIdentifierIsDefinedForManagedProjectAsync(
   }
 }
 
+export class AmbiguousBundleIdentifierError extends Error {
+  constructor(message?: string) {
+    super(message ?? 'Could not resolve bundleIdentifier.');
+  }
+}
+
 export async function getBundleIdentifierAsync(
   projectDir: string,
   exp: ExpoConfig,
-  { targetName, buildConfiguration }: { targetName?: string; buildConfiguration?: string } = {}
+  xcodeContext?: { targetName?: string; buildConfiguration?: string }
 ): Promise<string> {
   const workflow = await resolveWorkflowAsync(projectDir, Platform.IOS);
   if (workflow === Workflow.GENERIC) {
     warnIfBundleIdentifierDefinedInAppConfigForBareWorkflowProject(projectDir, exp);
 
-    const bundleIdentifier = IOSConfig.BundleIdentifier.getBundleIdentifierFromPbxproj(projectDir, {
-      targetName,
-      buildConfiguration,
-    });
+    const xcodeProject = IOSConfig.XcodeUtils.getPbxproj(projectDir);
+    const isMultiScheme = IOSConfig.BuildScheme.getSchemesFromXcodeproj(projectDir).length > 1;
+    const isMultiTarget = IOSConfig.Target.getNativeTargets(xcodeProject).filter(([, target]) =>
+      IOSConfig.Target.isTargetOfType(target, IOSConfig.Target.TargetType.APPLICATION)
+    );
```

Comment:
Sth is missing here (maybe `.length > 1`?). Right now, the type of `isMultiTarget` is `sth[]`.